umbral-pre/src/bindings_wasm.rs

```diff
@@ -0,0 +1,624 @@
+    pub fn random() -> Self {
+        Self {
+            backend: umbral_pre::SecretKey::random(),
+        }
+    }
+
+    /// Generates a secret key using the default RNG and returns it.
+    #[wasm_bindgen(js_name = publicKey)]
+    pub fn public_key(&self) -> PublicKey {
+        PublicKey {
+            backend: self.backend.public_key(),
+        }
+    }
+
+    #[wasm_bindgen(js_name = toSecretBytes)]
+    pub fn to_secret_bytes(&self) -> Box<[u8]> {
+        self.backend
+            .to_secret_array()
+            .as_secret()
+            .to_vec()
+            .into_boxed_slice()
+    }
+
+    #[wasm_bindgen(js_name = fromBytes)]
+    pub fn from_bytes(data: &[u8]) -> Result<SecretKey, JsValue> {
+        umbral_pre::SecretKey::from_bytes(data)
+            .map(|sk| Self { backend: sk })
+            .map_err(map_js_err)
+    }
+
+    #[allow(clippy::inherent_to_string)]
+    #[wasm_bindgen(js_name = toString)]
+    pub fn to_string(&self) -> String {
+        format!("{}", self.backend)
+    }
+}
+
+impl SecretKey {
+    pub fn inner(&self) -> &umbral_pre::SecretKey {
+        &self.backend
+    }
+}
+
+#[wasm_bindgen]
+pub struct SecretKeyFactory(umbral_pre::SecretKeyFactory);
+
+#[wasm_bindgen]
+impl SecretKeyFactory {
+    /// Generates a secret key factory using the default RNG and returns it.
+    pub fn random() -> Self {
+        Self(umbral_pre::SecretKeyFactory::random())
+    }
+
+    #[wasm_bindgen(js_name = seedSize)]
+    pub fn seed_size() -> usize {
+        umbral_pre::SecretKeyFactory::seed_size()
+    }
+
+    #[wasm_bindgen(js_name = fromSecureRandomness)]
+    pub fn from_secure_randomness(seed: &[u8]) -> Result<SecretKeyFactory, JsValue> {
+        umbral_pre::SecretKeyFactory::from_secure_randomness(seed)
+            .map(Self)
+            .map_err(map_js_err)
+    }
+
+    #[wasm_bindgen(js_name = makeKey)]
+    pub fn make_key(&self, label: &[u8]) -> Result<SecretKey, JsValue> {
+        self.0
+            .make_key(label)
+            .map(|sk| SecretKey { backend: sk })
+            .map_err(map_js_err)
+    }
+
+    #[wasm_bindgen(js_name = makeFactory)]
+    pub fn make_factory(&self, label: &[u8]) -> Self {
+        Self(self.0.make_factory(label))
+    }
+
+    #[wasm_bindgen(js_name = toSecretBytes)]
+    pub fn to_secret_bytes(&self) -> Box<[u8]> {
+        self.0
+            .to_secret_array()
+            .as_secret()
+            .to_vec()
+            .into_boxed_slice()
+    }
+
+    #[wasm_bindgen(js_name = fromBytes)]
+    pub fn from_bytes(data: &[u8]) -> Result<SecretKeyFactory, JsValue> {
+        umbral_pre::SecretKeyFactory::from_bytes(data)
+            .map(Self)
+            .map_err(map_js_err)
+    }
+
+    #[allow(clippy::inherent_to_string)]
+    #[wasm_bindgen(js_name = toString)]
+    pub fn to_string(&self) -> String {
+        format!("{}", self.0)
+    }
+}
+
+#[wasm_bindgen]
+#[derive(Serialize, Deserialize)]
```

Comment:
Are those for use in `nucypher_core` as well?

Reply:
Yes.